js/background.js

```diff
@@ -114,7 +114,20 @@ chrome.contextMenus.create(
 	}
 );
 
-var collections = JSON.parse(localStorage.getItem("sets")) || [];
+var collections = [];
```

Comment:
Ideally, we would have this react to changes in the storage though, like we did with the settings here. 

https://github.com/XFox111/TabsAsideExtension/blob/6267799cf0f7cdd72f6026e42b6cf835f4d4d540/js/aside-script.js#L89-L95
```
chrome.storage.onChanged.addListener((changes, namespace) =>
	{
		if (namespace == 'sync')
			for (key in changes)
				if (key === 'loadOnRestore')
					loadOnRestoreCheckbox.checked = changes[key].newValue
	});
```

I think it'd help avoid edge cases where chrome syncs some changes while the pane is open, and then the user saving a collection. This would cause us to **overwrite** these changes and completely replace the synced copy with the local one, leading to probable data loss.

Reply:
Yes please. Details are appreciated

Reply:
Okay, so let's say I have computer A at home, I have 1 collections there, and I've spent the weekend away with computer B, adding some other collections. Now I have 10 collections on computer B. It syncs, so the chrome backend has 10 collections has well.

I leave computer B and get back home, start my computer A , and chrome starts up. The extensions starts, and runs the background script, and lines 117-118 etc, once 
```
var collections = []
chrome.storage.sync.get("sets", values => ...)
```
It gets the collections from the sync storage, once. 
But what if chrome didn't sync fetch computer B's changes yet ? If it was busy updating something else, or if the internet access was still connecting.

Then this "collections" object on computer A contains 1 collection, and not 10. 
Now chrome manages to fetch the sync storage, and storage.sync contains 10 collections.

But the `collections` variable is *not* updated, and still contains 1 collection.

Now if I restore my single collection on my dear computer A, ``RestoreCollection`` is called. In RestoreCollections, I open the tabs, remove the collection from the `collections` variable, which now contains 0 collections. 
And finally, `chrome.storage.sync.set({ "sets": JSON.stringify(collections) })` replace the sync storage (that still contains 10 collections) with our local collections variable, that has 0 collections. The 9 collections I added on computer B are then gone and lost for good on the chrome backends as well.

This behavior is the same for any collections related operation, such as rename, create, or delete, currently.

Reply:
Yeah, I guess this can happen. Check out my [latest commit](https://github.com/XFox111/TabsAsideExtension/pull/43/commits/62b7fd9688f57b525bc873318fd9d64f5e94a851). This should fix it

Reply:
From experience, I feel like it's actually the most common scenario - the extension will be started up before sync happens, and sync may happen while the extension is running. 

I've spent some time debugging this, as this new commit causes more issues - could we move that sync feature to a separate PR, so it's decoupled from the 1.9 release ? It could be merged back in 1.9 later, or in 2.0 if you want to.